cookbooks/cdo-repository/libraries/git.rb

```diff
@@ -39,7 +39,7 @@ def fetch_updates
         run_opts = run_options(cwd: @new_resource.destination)
         # PATCH: support shallow fetch.
         converge_by("fetch updates for #{@new_resource.remote}") do
-          shell_out!("git fetch #{@new_resource.remote}#{" --depth #{@new_resource.depth}" if @new_resource.depth}", run_opts)
+          shell_out!("git fetch #{@new_resource.remote}", run_opts)
```

Comment:
It seems, by default, that fetch when the local repository is shallow is smart enough to only grab history this side of the 'shallow boundary' from the remote.  This should be safe to remove (unless we explicitly want to keep these at depth 1).

Resources:
- https://git-scm.com/docs/fetch-options#fetch-options---unshallow
- https://stackoverflow.com/questions/19352894/how-to-git-fetch-efficiently-from-a-shallow-clone